GEOSldas_App/util/postproc/ObsFcstAna_stats/postproc_ObsFcstAna.py

```diff
@@ -0,0 +1,392 @@
+
+# Tool to preprocess GEOSldas ObsFcstAna output into monthly sums, sums of squares, and sums of cross-products
+#
+# qliu, amfox, rreichle - May 2025
+
+import numpy as np
+import os
+
+import sys;       sys.path.append('../../shared/python/')
+
+import warnings;  warnings.filterwarnings("ignore")
+
+from datetime                    import datetime, timedelta
+from dateutil.relativedelta      import relativedelta
+from netCDF4                     import Dataset, date2num
+from read_GEOSldas               import read_ObsFcstAna, read_tilecoord, read_obs_param
+
+from helper.write_nc4            import write_sums_nc4, write_stats_nc4
+
+class postproc_ObsFcstAna:
+    
+    def __init__(self, exp_list, start_time, end_time, obs_from=0, sum_path='./', outid=None):
+        self.exp_list      = exp_list
+        self.expdir_list   = [item['expdir'] for item in exp_list]
+        self.expid_list    = [item['expid']  for item in exp_list]
+        self.exptag_list   = [item['exptag'] for item in exp_list]
+        self.domain        = exp_list[0]['domain']
+        self.start_time    = start_time 
+        self.end_time      = end_time
+        self.da_t0         = exp_list[0]['da_t0']
+        self.da_dt         = exp_list[0]['da_dt']
+        self.var_list      = ['obs_obs','obs_obsvar','obs_fcst','obs_fcstvar','obs_ana','obs_anavar']
+        self.tilecoord     = exp_list[0]['tilecoord']
+        self.obsparam_list = [item['obsparam'] for item in exp_list]
+        self.obs_from      = obs_from
+        self.sum_path      = sum_path
+
+        # If not provided, assemble filename of monthly sums, including info. of all experiments
+        if outid is None:
+            self.outid =  '_'.join([item for item in self.exptag_list])
+
+            if self.obs_from  > 0:
+                self.outid = self.outid + '_Obs_from_' + self.exptag_list[self.obs_from]
+    
+    # ----------------------------------------------------------------------------------------------------------
+    #
+    # Function to compute monthly sums from x-hourly ObsFcstAna data for a single month.
+
+    def compute_monthly_sums(self, date_time):
```

Comment:
I replaced the variable names *month with date_time for clarity.  If I understood correctly, the variable "month" included the full year/month/day/hour/min/sec info, so calling it "month" was very confusing.  I applied the same changes in several functions.  Please double-check carefully.  I hope I didn't mess it up.